org.OpenT2T.Sample.SuperPopular.Thermostat/Wink Thermostat/v2/thingTranslator.js

```diff
@@ -0,0 +1,117 @@
+"use strict";
+
+var q = require('q');
+var wh = require("opent2t-translator-helper-wink");
+
+// This code uses ES2015 syntax that requires at least Node.js v4.
+// For Node.js ES2015 support details, reference http://node.green/
+
+const EventEmitter = require("events");
+
+function validateArgumentType(arg, argName, expectedType) {
+    if (typeof arg === 'undefined') {
+        throw new Error('Missing argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + '.');
+    } else if (typeof arg !== expectedType) {
+        throw new Error('Invalid argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + ', got: ' + (typeof arg));
+    }
+}
+
+/**
+ * This translator class implements the "InterfaceA" interface.
+ * Since the interface includes a signal (notifiable property), this class
+ * extends from the node's built-in event-emitter class.
+ */
+class WinkThermostat extends EventEmitter {
+
+    constructor(deviceInfo) {
+        super(); // Construct EventEmitter base
+        this.initDevice(deviceInfo);
+    }
+
+    initDevice(deviceProps) {
+        console.log('Initializing device.');
+        var device = deviceProps;
+
+        validateArgumentType(device, 'device', 'object');
+        validateArgumentType(device.props, 'device.props', 'string');
+
+        var props = JSON.parse(device.props);
+        validateArgumentType(props.access_token, 'device.props.access_token', 'string');
+        validateArgumentType(props.id, 'device.props.id', 'string');
+       
+        var deviceId = props.id; 
+        var accessToken = props.access_token;
+
+        //use the winkHub device and initialize
+        wh.initWinkApi("thermostats",deviceId,accessToken);
+        console.log('Javascript and Wink Helper initialized : ' + deviceId + accessToken);
+    }
+
+    getPowerStatus() {
+        console.log('Get PowerStatus.');
+        return wh.getValueOfDesiredState('powered').then(isPowered => {
+            return new Promise((resolve, reject) => {
+                validateArgumentType(isPowered, "powered property", 'boolean');
+                resolve(isPowered ? "on" : "off");
+            });
+        });
+    }
+    
+    setPowerStatus(status) {
+        console.log('Set PowerStatus.');
+        validateArgumentType(status, 'Power status', 'string');
+
+        if(status.toUpperCase() === 'ON')
+            return wh.sendDesiredStateCommand('powered', true);
+        else
```

Comment:
Eventually I'd suggest we set up [jslint](https://github.com/reid/node-jslint) to run on all our translator code, to keep them clean and consistently formatted. One of the things it will enforce is braces around conditional blocks.